### PR TITLE
Cortx 30260: Added version compatibility check feature to upgrade script

### PR DESCRIPTION
**Pull request by kailashyadav11**
Signed-off-by: Kailash Yadav <kailash.yadav@seagate.com>

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds new functionality)
- [ ] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [ ] Documentation additions or improvements
- [ ] Code quality improvements to existing code or test additions/updates

## Applicable issues

- This change fixes an issue: #
- This change is related to an issue: #
- https://jts.seagate.com/browse/CORTX-30260

## CORTX image version requirements

This change requires the following images:

- `cortx-data:2.0.0-811
- `cortx-rgw:2.0.0-811
- `cortx-control:2.0.0-811

## Checklist

- [x] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [x] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change addresses a CORTX Jira issue:

- [x] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)

### CODE REVIEW

File: k8_cortx_cloud/upgrade-cortx-cloud.sh [hunk 1]
```diff
@@ -270,6 +282,77 @@ function initiate_upgrade() {
...
+    done
+    RULES="{\"requires\":[${RULES%?}]}"
+}
+
+function check_version_compatibility() {
+  echo -e "\n--------------------------------------"
+  while IFS= read -r line; do
+    echo -e "Checking Version Compatibility for $line\n"
+    HOSTNAME=$(hostname)
+    PORT="31169"
```
Comment (tanujashinde0405) [inline k8_cortx_cloud/upgrade-cortx-cloud.sh:333]: why port is hardcoded here? is it not coming from solution.yaml?

File: k8_cortx_cloud/upgrade-cortx-cloud.sh [hunk 2]
```diff
@@ -270,6 +282,77 @@ function initiate_upgrade() {
...
+    do
+      val=${val//[[:blank:]]/}
+      RULES+='"'$val'",'
+    done
+    RULES="{\"requires\":[${RULES%?}]}"
+}
+
+function check_version_compatibility() {
+  echo -e "\n--------------------------------------"
+  while IFS= read -r line; do
```
Comment (tanujashinde0405) [inline k8_cortx_cloud/upgrade-cortx-cloud.sh:330]: can we replce line with something like : deployment/pod/node?

File: k8_cortx_cloud/upgrade-cortx-cloud.sh [hunk 3]
```diff
@@ -270,6 +282,77 @@ function initiate_upgrade() {
...
+    echo -e "Checking Version Compatibility for $line\n"
+    HOSTNAME=$(hostname)
+    PORT="31169"
+    endpoint="https://${HOSTNAME}:${PORT}/api/v2/version/compatibility/node/${line}"
```
Comment (tanujashinde0405) [inline k8_cortx_cloud/upgrade-cortx-cloud.sh:334]: lets say endpoint as version_compatibility_endpoint or any meaningful name

File: k8_cortx_cloud/upgrade-cortx-cloud.sh [hunk 4]
```diff
@@ -270,6 +282,77 @@ function initiate_upgrade() {
...
+    response=$(curl -k -XPOST $endpoint -d ${RULES} -s | jq) 
+    echo "RESPONSE: $response"
```
Comment (tanujashinde0405) [inline k8_cortx_cloud/upgrade-cortx-cloud.sh:336]: do we need to show complete response the the user?

**Review by tanujashinde0405 — CHANGES_REQUESTED** [review]

**Review by nitin-seagate — APPROVED** [review]
LGTM

File: k8_cortx_cloud/upgrade-cortx-cloud.sh (hunk 1 above)
Comment (kailashyadav11) [reply k8_cortx_cloud/upgrade-cortx-cloud.sh]: changed to use solution.yaml

File: k8_cortx_cloud/upgrade-cortx-cloud.sh (hunk 2 above)
Comment (kailashyadav11) [reply k8_cortx_cloud/upgrade-cortx-cloud.sh]: done changed to node

File: k8_cortx_cloud/upgrade-cortx-cloud.sh (hunk 3 above)
Comment (kailashyadav11) [reply k8_cortx_cloud/upgrade-cortx-cloud.sh]: changed

File: k8_cortx_cloud/upgrade-cortx-cloud.sh (hunk 4 above)
Comment (kailashyadav11) [reply k8_cortx_cloud/upgrade-cortx-cloud.sh]: done removed the logging

**Review by tanujashinde0405 — APPROVED** [review]

**Pull request merged**